tasks/db_deploy/test/test_db_deploy.py

```diff
@@ -239,33 +253,585 @@ def return_cursor_return_function(conn: connection):
+        filename2 = uuid.uuid4().__str__()
+        os.environ[db_deploy.OS_ENVIRON_DDL_DIR_KEY] = ddl_dir + db_deploy.SEP
+
+        mock_get_file_names_in_dir.return_value = [filename1, filename2]
+        mock_con = Mock()
+        mock_get_db_connection.return_value = mock_con
+        mock_cursor = Mock()
+        mock_get_cursor.return_value = mock_cursor
+        os.environ[db_deploy.OS_ENVIRON_PLATFORM_KEY] = 'AWS'
+        mock_execute_sql_from_file.side_effect = [ResourceExists(Exception()), None]  # ResourceExists should NOT halt.
+
+        db_deploy.create_tables(db_host, db_name, db_port, db_user, db_password)
+
+        mock_get_cursor.assert_called_with(mock_con)
+        mock_execute_sql.assert_called_with(mock_cursor, """SET SESSION AUTHORIZATION dbo;""", mock.ANY)
+        mock_execute_sql_from_file.assert_has_calls([
+            call(mock_cursor, f"tables{db_deploy.SEP}{filename1}", mock.ANY),
+            call(mock_cursor, f"tables{db_deploy.SEP}{filename2}", mock.ANY)
+        ])
+        self.assertEqual(2, mock_execute_sql_from_file.call_count)
+
+        mock_con.close.assert_called()
+
+    @patch('db_deploy.walk_wrapper')
+    def test_get_file_names_in_dir_filters_out_init(self,
+                                                    mock_walk_wrapper: MagicMock):
+        """
+        Should not return init.sql.
+        """
+        directory = uuid.uuid4().__str__()
+        filename_1 = uuid.uuid4().__str__()
+        filename_2 = uuid.uuid4().__str__()
+
+        mock_walk_wrapper.return_value = [(None, None, [filename_1, 'init.sql', filename_2])]
+
+        result = db_deploy.get_file_names_in_dir(directory)
+
+        self.assertEqual({filename_1, filename_2}, set(result))
+        mock_walk_wrapper.assert_called_once_with(directory)
+
+    @patch('database.return_connection')
+    def test_get_db_connection_minimal_path(self,
+                                            mock_return_connection: MagicMock):
+        db_host = uuid.uuid4().__str__()
+        db_name = uuid.uuid4().__str__()
+        db_port = uuid.uuid4().__str__()
+        db_user = uuid.uuid4().__str__()
+        db_password = uuid.uuid4().__str__()
+
+        mock_con = Mock()
+        mock_return_connection.return_value = mock_con
+
+        result = db_deploy.get_db_connection(db_host, db_name, db_port, db_user, db_password)
+
+        self.assertIs(mock_con, result)
+        mock_return_connection.assert_called_once_with(
+            {"db_host": db_host,
+             "db_port": db_port,
+             "db_name": db_name,
+             "db_user": db_user,
+             "db_pw": db_password}
+        )
+
+    @patch('database.return_connection')
+    def test_get_db_connection_DbError_raises_DatabaseError(self,
+                                                            mock_return_connection: MagicMock):
+        db_host = uuid.uuid4().__str__()
+        db_name = uuid.uuid4().__str__()
+        db_port = uuid.uuid4().__str__()
```

Comment:
May want to use closer to valid data for port. port should be a number. For PostgreSQL default is 5432 but could use a random int converted to string.

Reply:
Makes sense. Fixed.